File: GUI_FCFA_converter.py
import tkinter as tk

# Function to convert EUR to FCFA
def convert():
    try:
        amount_eur = float(entry.get())
        amount_fcfa = amount_eur * 655.957  # Fixed exchange rate
        result_label.config(text=f"{amount_eur:.2f} EUR is equal to {amount_fcfa:.2f} FCFA")
    except ValueError:
        result_label.config(text="Please enter a valid number")

# Create the main window
root = tk.Tk()
root.title("Currency Converter")

# Create input label and entry
input_label = tk.Label(root, text="Enter amount in EUR:")
input_label.pack()

entry = tk.Entry(root)
entry.pack()

# Create convert button
convert_button = tk.Button(root, text="Convert", command=convert)
convert_button.pack()

# Create label for displaying result
result_label = tk.Label(root, text="")
result_label.pack()

# Run the application
root.mainloop()

File: GUI_FCFA_converter.R
library(shiny)

# Define UI
ui <- fluidPage(
  titlePanel("Currency Converter"),
  sidebarLayout(
    sidebarPanel(
      numericInput("amount_eur", "Enter amount in EUR:", value = 1, min = 0),
      actionButton("convert", "Convert")
    ),
    mainPanel(
      textOutput("result")
    )
  )
)

# Define server logic
server <- function(input, output) {
  observeEvent(input$convert, {
    amount_fcfa <- input$amount_eur * 655.957  # Fixed exchange rate
    output$result <- renderText({
      paste(input$amount_eur, "EUR is equal to", round(amount_fcfa, 2), "FCFA")
    })
  })
}

# Run the application
shinyApp(ui = ui, server = server)